runtime/bundles/org.eclipse.e4.core.di/src/org/eclipse/e4/core/internal/di/AnnotationLookup.java

```diff
@@ -42,14 +42,15 @@ public class AnnotationLookup {
 	private AnnotationLookup() {
 	}
 
-	public static record AnnotationProxy(List<Class<? extends Annotation>> classes) {
-		public AnnotationProxy {
-			classes = List.copyOf(classes);
+	static record AnnotationProxy(List<Class<? extends Annotation>> classes, List<String> classNames) {
```

Comment:
Has to be public to make `org.eclipse.e4.core.internal.services.MessageFactoryServiceImpl.processPostConstruct(Object, Class<?>)` compile.
- i would like to see this PR merged soon. Even if it may not be the final solution it still helps to solve the current situation

Reply:
Yes.

> Even if it may not be the final solution it still helps to solve the current situation

The problem with the first approach was that it only worked in multiple versions for some kind of annotations and not for all kind of annotations processed by the injector. And while partial progress is certainly better than no progress inconsistent behavior can be really bad.

But I think I have found a good approach now that works with all kind of annotation and I hope to complete it within the next days.